sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -70,10 +63,28 @@ public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
 
             if (liveVersions.Any())
             {
-                if (OnlyTestLatestServiceVersionLazy.Value)
+                if (TestEnvironment.GlobalTestOnlyLatestVersion)
                 {
                     _actualLiveServiceVersions = new[] { liveVersions.Max(Convert.ToInt32) };
                 }
+                else if (TestEnvironment.GlobalTestServiceVersions.Any())
```

Comment:
nit to avoid Linq having to inspect if this is a Collection type: 
```suggestion
                else if (TestEnvironment.GlobalTestServiceVersions.Length > 0)
```